### ISSUE TITLE
Updating system gives me a clean database, all data lost.

### ISSUE THREAD

**Issue by MarkLFT** [NginxProxyManager/nginx-proxy-manager#3070]
I am trying to update my npm installation.  I currently have version 2.9.19 with MariaDB version. 10.4.12.

I used 
docker-compose down
docker-compose pull  -- for the database it says error, then app pulls the latest version.
docker-compose up -- I can see it is trying to migrate the database, and it creates a new admin account. 

When I try to login using my previous admin account it says not found, if I log in with the newly created account, I get in, but all my hosts and streams are gone.

How can I update my installation with losing all my data?

**Comment by vodjan** [comment]
Same issue here, currently on NPM v2.10.3 and MariaDB-Aria 3.12.0.

I can no longer log into NPM with my previous account info, but I can log in using the default credentials and none of my configs show up when I do. All of my reverse proxies still function just fine even though they aren't visible in the web GUI.

The DB runs in a separate container, both are configured with one compose file. No other container uses this DB instance. I manage both through Portainer, where I update via recreating the containers while pulling the latest image. I have been doing this for at least a year and a half and never had an issue.

I can't pinpoint which exact update caused this because I've only been checking NPM functionality by connecting to the reverse proxies for some time now, and they still work fine.

**Comment by farhan-tariq** [comment]
Why not have a separate docker-compose.yml file specific for running a database. Might run into docker network issue but they are resolvable as you can make the database and npm be on the same network.

**Comment by MarkLFT** [comment]
> Why not have a separate docker-compose.yml file specific for running a database. Might run into docker network issue but they are resolvable as you can make the database and npm be on the same network.
> 
> So when you will do docker-compose down, it will not bring the database down. The issue might be that the data is not getting persisted on your host OS

I am not sure I understand th point you are making.  The docker compose is as per recommended by the development team, and the issue has nothing to do with the database server running or not, it is more to do with the database data appears to get lost during the update.

**Comment by vodjan** [comment]
> Why not have a separate docker-compose.yml file specific for running a database.

I doubt that makes a difference here.

> The issue might be that the data is not getting persisted on your host OS

Well for me it's definitely getting persisted in some way, because as I said the configuration continues to work after both `docker-compose down` and machine reboots. I just can't see it in the UI and my account got lost.

**Comment by GloriousGloria23** [comment]
yep, same here... interestingly, i find a populated database.sqlite in the docker container compose folder... but npm is not accessing it... anyone has an idea how to repopulate it?\

**Comment by R1skNt** [comment]
I just encountered the same issue, stop>pull>run all data is gone, even tho the PostgreSQL data volume is there

**Comment by ProtossGP32** [comment]
Hi all! Same issue here, with Docker compose, when I `down` and `up` again, NPM dashboard asks me again to create an admin account and no proxy hosts are shown although they appear in the Docker volume. Any hint on how to solve this? PostgreSQL volume is completely empty.

EDIT: I've found out that postgresql data wasn't being properly written in the mounted volume. Even though `/var/lib/postgresql` path was mounted in a local `./postgresql`, the container folder `/var/lib/postgresql/data` wasn't being created in my local filesystem. I've solved it by mounting `/var/lib/postgresql/data` instead:

```yaml
  db:
    image: postgres:17
    environment:
      POSTGRES_USER: 'npm'
      POSTGRES_PASSWORD: 'npmpass'
      POSTGRES_DB: 'npm'
    volumes:
      - ./postgresql_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -d npm -U npm"]
      interval: 10s
      timeout: 5s
      retries: 5
```
Doing this persists the postgresql data between rollouts.
I've looked at other postgresql docker-compose.yaml examples in the web and all of them seem to mount the data path instead of the parent one.
